src/java/main/org/apache/zookeeper/server/SessionTrackerImpl.java

```diff
@@ -84,6 +84,9 @@ public static long initializeNextSession(long id) {
         long nextSid;
         nextSid = (Time.currentElapsedTime() << 24) >>> 8;
         nextSid =  nextSid | (id <<56);
+        if (nextSid == EphemeralType.CONTAINER_EPHEMERAL_OWNER) {
+            ++nextSid;  // this is an unlikely edge case, but check it just in case
```

Comment:
This makes me nervous but I can't think of a reason why it wont work. It may be worth changing the comment on this method because it is no longer strictly true.

Reply:
"no longer strictly true" - actually it still is true. If `Time.currentElapsedTime()` returns 0 and server Id is 0x80, this will happen. That's why nextSid has to be incremented in that case.